### PR TITLE
Update  5.1 ( Skipped $banID )

### PR DESCRIPTION
**Pull request by Rainb0wKey**
YES

### CODE REVIEW

File: src/bdscript/blackListRolesIDs.md [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-$blackListRolesIDs[roleIDs;errorMessage]
+$blackListRolesIDs[role IDs;...;error message]
 ```
 
-### Breakdown
-- `roleIDs` - The roles that will be blacklisted. Separate roleIDs using `;`.
-- `errorMessage` - The message that sends if the user has a role in the blacklist.
+### Parameters
+- `role IDs`: The roles that will be blacklisted. Separate role IDs using `;`.
+- `error message`: The message that sends if the user has a role in the blacklist.
```
Comment (MineBartekSA) [inline src/bdscript/blackListRolesIDs.md]:
```suggestion
- `error message`: The message that will be sent if the user has a role from the blacklist.
```

File: src/bdscript/blackListRoles.md [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
-$blackListRoles[roleNames;errorMessage]
+$blackListRoles[role names;...;error message]
 ```
 
-### Breakdown
-- `roleNames` -  The name(s) of the role(s) to blacklist. Separate role names using `;`.
-- `errorMessage` - The message that appears when the user has a role in the blacklist.
+## Parameters
+- `role names`: The name(s) of the role(s) to blacklist. Separate role names using `;`.
+- `error message`: The message that appears when the user has a role in the blacklist.
```
Comment (MineBartekSA) [inline src/bdscript/blackListRoles.md]:
```suggestion
- `error message`: The message that will be sent if the user has a role from the blacklist.
```

File: src/bdscript/blackListIDs.md [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
 ```
 
-### Breakdown
-- `userIDs` - The users to blacklist from using the command. Separate userIDs with `;`.
-> 🧙‍♂️ [Please provide real userID(s). Click-me for more info!](https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-)
-- `errorMessage` - The message that appears when the user running the command is blacklisted.
+## Parameters
+- `user IDs`: The users to blacklist from using the command. Separate user IDs with `;`.
+> [How to get user ID?](https://support.discord.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-)
+- `error message`: The message that appears when the user running the command is blacklisted.
```
Comment (MineBartekSA) [inline src/bdscript/blackListIDs.md]:
```suggestion
- `error message`: The message that will be sent when the user running the command is blacklisted.
```

File: src/bdscript/banID.md [hunk 4]
```diff
@@ -1,38 +1,25 @@
...
-### Usage #3
-*(recommended)*
-
-```
-$banID[reason;userID]
-```
-Gets the user to ban from the 'userID' field. The 'reason' will show up in audit logs. You can also leave the 'reason' field empty to provide no reason.
 
+## Parameters
+- `reason`: The reason for this action, which will be sent to the audit-log. _(Optional)_
```
Comment (MineBartekSA) [inline src/bdscript/banID.md]:
```suggestion
- `reason`: The reason for this action, which will be saved in the audit log. Can be left empty.
```

File: src/bdscript/banID.md [hunk 5]
```diff
@@ -1,38 +1,25 @@
...
+> The reason for the ban can be viewed using `$getBanResaon[]`.
+- `user ID`: Gets the user to ban. _(Optional)_
```
Comment (MineBartekSA) [inline src/bdscript/banID.md]:
```suggestion
- `user ID`: The user to ban. _(Optional)_ Can be omitted, then the ID will be taken from the last part of the command message.
```

File: src/bdscript/banID.md [hunk 6]
```diff
@@ -1,38 +1,25 @@
...
 Bans a user using their ID.
 
-## Usages
-There are three usages of the `$banID` function.
 
-### Usage #1
+## Syntax
 ```
-$banID
+$ban[(reason;user ID)]
```
Comment (MineBartekSA) [inline src/bdscript/banID.md]:
```suggestion
$banID[reason;(user ID)]
```

File: src/bdscript/ban.md [hunk 7]
```diff
@@ -1,29 +1,29 @@
...
+## Syntax
 ```
-$ban[(optional) reason]
+$ban[(reason)]
 ```
 
-### Breakdown
-- `reason` - The reason for this action, which will be sent to the audit-log. *(Optional)*
+## Parameters
+- `reason`: The reason for this action, which will be sent to the audit-log. _(Optional)_
```
Comment (MineBartekSA) [inline src/bdscript/ban.md]:
```suggestion
- `reason`: The reason for this action, which will be save in the audit log. Can be left empty.
```

**Review by MineBartekSA — CHANGES_REQUESTED** [review]
I want you to redo the `ban` and `banID` pages.
I did some comments, but they are mostly just spelling mistakes and such.
I want you to highlight the simple functions in them too.

Because both `ban` and `banID` are Simple **and** Complex functions.
They work differently from each other, so they need to be better explained.
You can quickly look up their in-app descriptions here: [`$ban`](https://botdesignerdiscord.com/public/function/$ban), [`$ban[]`](https://botdesignerdiscord.com/public/function/$ban[]), [`$banID`](https://botdesignerdiscord.com/public/function/$banID), and [`$banID[]`](https://botdesignerdiscord.com/public/function/$banID[]).
(As a side note, we [the developers] need to make some of these in-app descriptions more verbose)

You also made some serious mistakes with arguments too.
The `reason` argument is always required but can be left empty in both functions.

**Comment by Rainb0wKey** [comment]
For all page changes. I only changed the description of the arguments 2 times. They were like this until my change.

**Comment by Rainb0wKey** [comment]
My understanding is that I will need to create 2 more pages for these functions? with other arguments?

**Comment by MineBartekSA** [comment]
The current design of the wiki doesn't allow for separate pages for simple and complex functions with the same name.
At least without renaming a whole lot of pages.
I was thinking of just keeping it how it is right now.

Pages are titled with the "base" name and they cover every function with that name like [`$dm`](https://nilpointer-software.github.io/bdfd-wiki/nightly/bdscript/dm.html) or the current [`$banID`](https://nilpointer-software.github.io/bdfd-wiki/nightly/bdscript/banID.html) does.
And since you are rewriting those pages, you could try to create a new and better way to address simple and complex functions on the same page.

I'd say that the current way pages like this are structured isn't that great.
They especially don't highlight that technically `$dm` and `$dm[]` are two different functions.
Sure, most people don't ever need to know about this, but it would hurt to make some sort of distinction between them.
Especially **if** they differ significantly in behaviour. Thankfully, most functions aren't like that.

I'm leaving it to you how you end up writing those pages.
The only thing I'd like you to not forget is that there are simple and complex functions with the same names.

**Comment by DevSpen** [comment]
Although it doesn’t necessarily need to addressed in this pull request, I think we need to update SUMMARY.md and some wiki pages to represent functions the same way the app does. The more coherent the better.
- Functions without brackets would be shown as all functions are currently are (`$functionName`)
- Functions with brackets would be shown with brackets (`$functionName[]`)
- "Analogous function[s]"—functions w/ same name but one has brackets and the other doesn’t—would have separate pages regarding them.
    - There would be a note on each page linking to the other page on the respective function (i.e `$functionName`'s wiki would have a note saying there is another function `$functionName[]`)

@MineBartekSA (Referring to the in-app function list) On the developer end, how do you guys link a function within the app to its respective wiki page? As of right now, it doesn’t include brackets no matter if the function has brackets or not (which is how it should work based on the current design). Also, some functions (like the ones relating to buttons) only link to the Buttons guide within the wiki and not the specific function—is that pre-programmed behavior? Could things like that be changed smoothly though?

**Comment by MineBartekSA** [comment]
To have an easier transition and to help people distinguish simple and complex functions, we could split the `BDScript` section in two, like so:

- BDScript
  - Simple functions
    - $message
  - Complex functions
    - $message[]

Then pages and URLs wouldn't need to include `[]` in them, which would probably help in some cases.

@DevSpen The app, as you'd expect, just trims the function name. And then there are explicitly chosen functions that open guides instead of their function wikis. To be honest, we probably should either add another button with `Open guide` for them.

Also, if we are going to reorganise the wiki in some way, we should probably move this discussion [somewhere else](https://github.com/NilPointer-Software/bdfd-wiki/discussions/254).

**Review by MineBartekSA — APPROVED** [review]
Thank you for your hard work

**Pull request merged**